### ISSUE TITLE
firebase_database plugin example has setState warning

### ISSUE THREAD

**Issue by mit-mit** [flutter/flutter#12257]
## Steps to Reproduce

```
git clone 
cd packages/firebase_database/example/
flutter run
hit the + button a few times
```

get this:

```
#0      State.setState.<anonymous closure> (package:flutter/src/widgets/<…>
setState() called after dispose(): FirebaseAnimatedListState#8e381(lifecycle state: defunct, not mounted)
This error happens if you call setState() on a State object for a widget that no longer appears in the widget tree (e.g., whose parent widget no longer includes the widget in its build). This error can occur when code calls setState() from a timer or an animation callback. The preferred solution is to cancel the timer or stop listening to the animation in the dispose() callback. Another solution is to check the "mounted" property of this object before calling setState() to ensure the object is still in the tree.
This error might indicate a memory leak if setState() is being called because another object is retaining a reference to this State object after it has been removed from the tree. To avoid memory leaks, consider breaking the reference to this object during dispose().
```

**Comment by mit-mit** [comment]
cc @kroikie @collinjackson

**Comment by collinjackson** [comment]
This looks like a valid issue; we should re-test after flutter/plugins#101 lands to see if that fixed it or not.

**Comment by branflake2267** [comment]
I have a another solution for this. The subscriptions has to be canceled after the firebase animated list is disposed. The subscriptions are walled in the list, so I used clear to hook the cancel subscriptions. 
https://github.com/flutter/plugins/pull/268

**Comment by collinjackson** [comment]
Looks like this was fixed in flutter/plugins#268

**Comment by kroikie** [comment]
@mit-mit

This issue has been moved to https://github.com/FirebaseExtended/flutterfire/issues/965. Any further collaboration will be done there.
